r/src/config.cpp

```diff
@@ -21,6 +21,13 @@
 
 #include <arrow/config.h>
 
+// [[arrow::export]]
+std::vector<std::string> build_info() {
+  auto info = arrow::GetBuildInfo();
+  return {info.version_string, info.compiler_id, info.compiler_version,
+          info.compiler_flags, info.git_id};
```

Comment:
What happens if I'm not building from a git checkout (like from the apache source tarball or something)?

Reply:
Empty string